src/mlpack/methods/ann/layer/lambda_map_reduce.hpp

```diff
@@ -0,0 +1,270 @@
+/**
+ * @file methods/ann/layer/lambda_map_reduce.hpp
+ * @author Shubham Agrawal
+ *
+ * Base class for neural network layers that are wrappers around other layers.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_LAMBDA_MAP_REDUCE_HPP
+#define MLPACK_METHODS_ANN_LAYER_LAMBDA_MAP_REDUCE_HPP
+
+#include "../make_alias.hpp"
+#include "layer.hpp"
+#include <mlpack/methods/ann/reduction_rules/add_reduction.hpp>
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * A lambda "map-reduce" is a layer that is a wrapper around other layers.  
```

Comment:
Why is the `Lambda` part of the name?  I don't get that part.

Reply:
I think I can remove lambda here as I am not using any lambda fn to specify layers.